### DISCUSSION TITLE
Recipe versioning

### DISCUSSION THREAD

**Discussion by rafaelmathieu** [mealie-recipes/mealie#880] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
I have a lot of recipes that have slight variation to them based on who I am making it for and maybe the event.
Currently If I want to keep track on this, I would have to either:

- Create separate recipe for all variations
- Add comments to keep track of the differences
- edit the recipe steps to have all the possible alternates

All of those options are not very straight forward to deal with.

Example 1: I have 3 recipes for the same rum cake. What is different is that each one is for a specific mold that I use, and because of this they have vastly different ingredients requirements and cooking instructions. Currently I have those as three different recipes, but it looks odd to have all three show up on the main page.

**Suggest A Solution**
When creating or editing a recipe, it would be great to be able to point them to a parent recipe. This way it would all be neat and organized. Maybe have an overall option to allow them to show in the regular view if users wants this.

When you go to the parent recipe it would show you all the child recipe available. 

* Details of the technical implementation
* Tradeoffs made in design decisions
* Caveats and considerations for the future

If there are multiple solutions, please present each one separately. Save comparisons for the very end.)

**Additional Information**
_FILL THESE OUT_
- [x] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by DrAmygdala** [comment]
Is this a feature might get implemented? I'd love this feature and would be happy to help if it's approved.

**Comment by michael-genson** [comment]
Similar discussion here: https://github.com/mealie-recipes/mealie/discussions/1761
